module_uninstall_check/wizards/wizard_module_uninstall.py

```diff
@@ -0,0 +1,117 @@
+        multi='module_ids',
+        comodel_name='ir.module.module', readonly=True,
+        help="Modules list that will be uninstalled by dependency")
+
+    module_qty = fields.Integer(
+        string='Impacted modules Quantity', compute='_compute_module_ids',
+        multi='module_ids', readonly=True)
+
+    module_name = fields.Char(
+        string='Module Name', related='module_id.name', readonly=True)
+
+    model_line_ids = fields.One2many(
+        comodel_name='wizard.module.uninstall.line', readonly=True,
+        inverse_name='wizard_id', domain=[('type', '=', 'model')])
+
+    field_line_ids = fields.One2many(
+        comodel_name='wizard.module.uninstall.line', readonly=True,
+        inverse_name='wizard_id', domain=[('type', '=', 'field')])
+
+    # Compute Section
+    @api.multi
+    @api.depends('module_id')
+    def _compute_module_ids(self):
+        for wizard in self:
+            if wizard.module_id:
+                res = wizard.module_id.downstream_dependencies()
+                wizard.module_ids = res
+                wizard.module_qty = len(res)
+            else:
+                wizard.module_ids = False
+                wizard.module_qty = 0
+
+    # OnChange Section
+    @api.multi
+    @api.onchange('module_id')
+    def onchange_module_id(self):
+        model_data_obj = self.env['ir.model.data']
+        model_obj = self.env['ir.model']
+        field_obj = self.env['ir.model.fields']
+
+        for wizard in self:
+            wizard.model_line_ids = False
+            wizard.field_line_ids = False
+
+        for wizard in self.filtered(lambda x: x.module_id):
+            model_ids = []
+            module_names = wizard.module_ids.mapped('name')\
+                + [wizard.module_id.name]
+            # Get Models
+            models_data = []
+            model_xml_ids = model_data_obj.search([
+                ('module', 'in', module_names),
+                ('model', '=', 'ir.model')]).mapped('res_id')
+            model_xml_ids = list(set(model_xml_ids))
+            for model in model_obj.browse(model_xml_ids).filtered(
+                    lambda x: not x.osv_memory):
+                # Filter models that are not associated to other modules,
+                # and that will be removed, if the selected module is
+                # uninstalled
+                other_declarations = model_data_obj.search([
+                    ('module', 'not in', module_names),
+                    ('model', '=', 'ir.model'),
+                    ('res_id', '=', model.id)])
+                if not len(other_declarations):
+                    models_data.append((0, 0, {
+                        'type': 'model',
+                        'model_id': model.id,
+                    }))
+                    model_ids.append(model.id)
+            wizard.model_line_ids = models_data
+
+            # Get Fields
+            fields_data = []
+            field_xml_ids = model_data_obj.search([
```

Comment:
Rename `field_xml_ids` to `field_ids` because it does store database ids, not xml_ids.